bindings/nodejs/lib/types/block/core/block.ts

```diff
@@ -1,67 +1,206 @@
 // Copyright 2023 IOTA Stiftung
 // SPDX-License-Identifier: Apache-2.0
 
-import { BasicBlock } from './basic';
-import { ValidationBlock } from './validation';
+import { AccountId } from '../id';
+import { Signature, SignatureDiscriminator } from '../signature';
+import { SlotCommitmentId, SlotIndex } from '../slot';
+import { u64 } from '../../utils/type-aliases';
+import { plainToInstance, Type } from 'class-transformer';
+import { BlockBody } from './block-body';
+import { BasicBlockBody } from './basic';
+import { ValidationBlockBody } from './validation';
+import { BlockBodyDiscriminator } from '.';
+
 /**
- * All of the block types.
+ * Represent the object that nodes gossip around the network.
  */
-export enum BlockType {
-    /// A Basic block.
-    Basic = 0,
-    /// A Validation block.
-    Validation = 1,
-}
+class Block {
```

Comment:
This is missing a proper `header` for the serialisation?

Reply:
Changed